### DISCUSSION TITLE
[FEATURE REQUEST] Continuous fuzzing

### DISCUSSION THREAD

**Discussion by AdamKorcz** [saltstack/salt#67368] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I hereby suggest setting up continuous fuzzing of Salt.

**Describe the solution you'd like**
Googles OSS-fuzz projects recently added support for Python fuzzing. I have been working on integrating Salt into OSS-fuzz and the integration can be found here: https://github.com/google/oss-fuzz/pull/4948. It includes a fuzzer for the yaml utils.

When integrated, all fuzzers will be run continuously by OSS-fuzz free of charge to look for hard-to-find bugs in the codebase. Adding new fuzzers is easy and can easily be included in [the build script](https://github.com/google/oss-fuzz/pull/4948/files#diff-930964f038b6ecaac7f791f52834d3b6356d8cde1976a0296e340e40d1b46454).

For those unfamiliar: Fuzzing is a method for testing applications whereby pseudo-random data is passed to the target application with the goal of finding bugs and vulnerabilities. Over the course of OSS-fuzz's lifetime some critical bugs have been found in projects written in both memorysafe and -unsafe languages, and bugs have been found in Python projects as well.

It is free for critical open source projects to integrate, however with the implied expectation that bugs are fixed so that resources go towards resolving the bugs.

Although not necessary, I do suggest moving the build script and the fuzzers to Salts own repository to easier modify these. For now, however, they can also remain hosted at OSS-fuzz's repository.

To complete the integration at least one maintainers email address is needed in the project.yaml file. I have left my own email address in there as well to see the integration through to completion, but if you'd rather have me off that list, do let me know and I will be happy to remove it. All maintainers have access to bug reports which means that if any bugs and vulnerabilities come up, I will be able to see them.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by AdamKorcz** [comment]
As mentioned on the PR over at OSS-fuzz, the current fuzzer mainly targets code in pyyaml that is already being fuzzed. There are a lot of other good targets in Salt, and I will be happy to write a few more fuzzers for the initial integration. Before doing so it would be great to hear if there is interest in continuous fuzzing of Salt at all as well as fixing potential bugs that are found in the process.
